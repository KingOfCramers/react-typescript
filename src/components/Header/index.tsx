import React, { useState } from "react";
import { Drawer, IconButton } from "@material-ui/core";
import { Menu, Close } from "@material-ui/icons";
import history from "../../history";

import Links from "./links";

const Header = React.memo(function Header(props) {
  const [mobileOpen, setMobileOpen] = useState(false);

  const handleGoHome = () => {
    history.push("/");
  };

  function handleDrawerToggle() {
    setMobileOpen(!mobileOpen);
  }

  return (
    <header className="header shadow">
      <nav>
        <IconButton onClick={handleDrawerToggle} className="menuButton pointer">
          <Menu />
        </IconButton>
        <h1 onClick={handleGoHome} className="mainTitle pointer">
          Cloture
        </h1>
        <Drawer
          className="drawer"
          variant="temporary"
          anchor={"left"}
          open={mobileOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true, // Better open performance on mobile.
          }}
        >
          <IconButton className="close" onClick={handleDrawerToggle}>
            <Close />
          </IconButton>
          <Links
            setMobileOpen={setMobileOpen}
            links={[
              { label: "Calendar", link: "/calendar" },
              { label: "Senate", link: "/committees/senate" },
              { label: "House", link: "/committees/house" },
            ]}
          />
        </Drawer>
      </nav>
    </header>
  );
});

export default Header;
