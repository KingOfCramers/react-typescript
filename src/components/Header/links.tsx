import React, { ReactElement } from "react";
import { ListItem } from "@material-ui/core";
import history from "../../history";

interface Link {
  label: string;
  link: string;
}
export interface Props {
  links: Link[];
  setMobileOpen(arg: boolean): void;
}

export default function Links({
  links,
  setMobileOpen,
}: Props): ReactElement | null {
  const handleGoToLink = (link: string): void => {
    setMobileOpen(false);
    history.push(link);
  };

  return (
    <ul>
      {links.map(({ link, label }) => (
        <ListItem button onClick={() => handleGoToLink(link)}>
          <li className="pointer">
            <p className="menu plain tight">{label}</p>
          </li>
        </ListItem>
      ))}
    </ul>
  );
}
