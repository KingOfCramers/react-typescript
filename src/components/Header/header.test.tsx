import React from "react";
import { shallow } from "enzyme";
import Header from "./index";

describe("Header", () => {
  it("Should render the Header", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
