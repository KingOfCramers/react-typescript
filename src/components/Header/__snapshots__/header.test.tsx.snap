// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render the Header 1`] = `
<header
  className="header shadow"
>
  <nav>
    <WithStyles(ForwardRef(IconButton))
      className="menuButton pointer"
      onClick={[Function]}
    >
      <MenuIcon />
    </WithStyles(ForwardRef(IconButton))>
    <h1
      className="mainTitle pointer"
      onClick={[Function]}
    >
      Cloture
    </h1>
    <WithStyles(ForwardRef(Drawer))
      ModalProps={
        Object {
          "keepMounted": true,
        }
      }
      anchor="left"
      className="drawer"
      onClose={[Function]}
      open={false}
      variant="temporary"
    >
      <WithStyles(ForwardRef(IconButton))
        className="close"
        onClick={[Function]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
      <Links
        links={
          Array [
            Object {
              "label": "Calendar",
              "link": "/calendar",
            },
            Object {
              "label": "Senate",
              "link": "/committees/senate",
            },
            Object {
              "label": "House",
              "link": "/committees/house",
            },
          ]
        }
        setMobileOpen={[Function]}
      />
    </WithStyles(ForwardRef(Drawer))>
  </nav>
</header>
`;
